common/primitives/src/msa.rs

```diff
@@ -65,6 +65,19 @@ where
 	pub schema_permissions: BoundedBTreeMap<SchemaId, BlockNumber, MaxSchemaGrantsPerDelegation>,
 }
 
+// Cannot derive the PartialEq without a mess of impl PartialEq for MaxSchemaGrantsPerDelegation
+impl<SchemaId, BlockNumber, MaxSchemaGrantsPerDelegation> PartialEq
+	for Delegation<SchemaId, BlockNumber, MaxSchemaGrantsPerDelegation>
+where
+	SchemaId: PartialEq,
+	BlockNumber: PartialEq,
+	MaxSchemaGrantsPerDelegation: Get<u32>,
+{
+	fn eq(&self, other: &Self) -> bool {
+		self.revoked_at == other.revoked_at && self.schema_permissions == other.schema_permissions
```

Comment:
By adding this one impl here, we can remove a ton of impl elsewhere

Reply:
It's nice being able to do a deep eq check on `BoundedBTreeMap` in Rust by just `==`. In some other languages this would not be as easy.